sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ReviewSqlQueriesForSecurityVulnerabilities.cs

```diff
@@ -55,6 +62,28 @@ public sealed class ReviewSqlQueriesForSecurityVulnerabilities : SonarDiagnostic
             KnownType.System_Data_SqlClient_SqlDataAdapter
         };
 
+        private string protobufDirectory;
+        private int protobufFileIndex = 0;
+
+        /// <summary>
+        /// Contains the build configuration name as set by Scanner for MSBuild. Usually it
+        /// is a number. We include this in the protobuf file names because the Sonar Security
+        /// plugin is unable to read files from subfolders.
+        /// </summary>
+        private string configurationName;
```

Comment:
I found this name really confusing, even with the comment above. I'm trying to work out why that is :-) I think it's because "build configuration" generally means _debug_ or _release_ in MSBuild, and also because it's more like an id than a name.
How about _projectBuildId_ - _a unique identifier generated by the Scanner for MSBuild for this invocation of the compiler for the current project_?